### PR TITLE
Build the project structure

### PR DESCRIPTION
**Pull request by ghost**
Task description:
 - Add the spring boot project skeleton with Maven
 - Add Dockerfile for building image and run the container 
 - Include the helloworld api execution.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-actuator</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
```
Comment (caominhnhut) [inline pom.xml:40]: add space here

File: src/main/java/com/hipo_delivery_be/App.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+@SpringBootApplication
+@RestController
+public class App {
+    public static void main(String[] args) {
+        SpringApplication.run(App.class, args);
+    }
+
+    @RequestMapping(value = "/api/helloworld")
+    ResponseEntity<?> hello() {
```
Comment (caominhnhut) [inline src/main/java/com/hipo_delivery_be/App.java]: Move to HeathCheckController

File: src/main/java/com/hipo_delivery_be/App.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        Object resBody = new Object() {
```
Comment (caominhnhut) [inline src/main/java/com/hipo_delivery_be/App.java]: Just return a String will be OK with me

File: src/main/docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,11 @@
+FROM eclipse-temurin:17-jre-alpine
```
Comment (caominhnhut) [inline src/main/docker/Dockerfile]: move the docker folder to root directory

File: pom.xml (hunk 1 above)
Comment (ghost) [reply pom.xml:40]: @caominhnhut Added space to new Controller Class on latest commit

File: src/main/java/com/hipo_delivery_be/App.java (hunk 2 above)
Comment (ghost) [reply src/main/java/com/hipo_delivery_be/App.java]: @caominhnhut Moved to new Controller Class on latest commit

File: src/main/docker/Dockerfile (hunk 4 above)
Comment (ghost) [reply src/main/docker/Dockerfile]: @caominhnhut Moved to root directory on latest commit

File: src/main/java/com/hipo_delivery_be/App.java (hunk 3 above)
Comment (ghost) [reply src/main/java/com/hipo_delivery_be/App.java]: @caominhnhut Changed return type to String on latest commit

File: src/main/java/com/hipo_delivery_be/App.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package com.hipo_delivery_be;
```
Comment (caominhnhut) [inline src/main/java/com/hipo_delivery_be/App.java]: com.hipo.delivery

File: src/main/java/com/hipo_delivery_be/App.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@SpringBootApplication
+public class App {
```
Comment (caominhnhut) [inline src/main/java/com/hipo_delivery_be/App.java]: HipoDeliveryApplication

File: src/main/java/com/hipo_delivery_be/controller/HealthCheckController.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package com.hipo_delivery_be.controller;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class HealthCheckController {
+    @RequestMapping(value = "/api/helloworld")
```
Comment (caominhnhut) [inline src/main/java/com/hipo_delivery_be/controller/HealthCheckController.java]: @GetMapping

File: src/main/resources/application.yml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+# See http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html
+spring:
+  thymeleaf.cache: false
```
Comment (caominhnhut) [inline src/main/resources/application.yml]: remove since we don't use it

File: src/main/java/com/hipo_delivery_be/App.java (hunk 6 above)
Comment (ghost) [reply src/main/java/com/hipo_delivery_be/App.java]: @caominhnhut Updated to HipoDeliveryApplication on new commit

File: src/main/java/com/hipo_delivery_be/controller/HealthCheckController.java (hunk 7 above)
Comment (ghost) [reply src/main/java/com/hipo_delivery_be/controller/HealthCheckController.java]: @caominhnhut Changed to @GetMapping on new commit

File: src/main/resources/application.yml (hunk 8 above)
Comment (ghost) [reply src/main/resources/application.yml]: @caominhnhut Removed on new commit

**Review by caominhnhut — APPROVED** [review]

**Review by sts-nhutnguyen — APPROVED** [review]

**Pull request merged**